FirebaseTestProject/Models/Post.swift

```diff
@@ -49,13 +79,37 @@ struct PostService {
         return db.collection("posts")
     }
     
+    static var favoritesReference: CollectionReference {
+        let db = Firestore.firestore()
+        return db.collection("favorites")
+    }
+    
     static func getPosts() async throws -> [Post] {
         let postsSnapshots = try await postsReference.getDocuments()
         let posts = postsSnapshots.documents.map { Post(from: $0.data()) }
         return posts
     }
     
+    static func getFavorites() async throws -> [Post] {
+        let userid = UserDefaults.standard.value(forKey: "userid") != nil ?  UserDefaults.standard.value(forKey: "userid") as! String : "00854E9E-8468-421D-8AA2-605D8E6C61D9"
```

Comment:
Generally speaking, we are going to have to save a UserID during authentication, though. And, we would only ever need to get a favorite for the current user. I would think that we would automatically call userID inside the function since it isn't ever going to change per a parameter.

Reply:
Be it UserDefaults or some other method, I just assumed that we would need to store the value. That could be a false assumption.

Reply:
Fair enough! I’d say it’s less a matter of whether the value is going to be changed or saved, and more a matter of how we want to architect our codebase.

We need to retrieve the user ID somehow, but how do we want to do that? I suggested passing it as a parameter because the `PostService` is for posts, not for digging up user information. By keeping our services as focused as possible, we make them easier to test, maintain, and reason about. It would also make the API more explicit in declaring its data dependencies, which also makes our code easier to understand and maintain.

That being said, I’m definitely sympathetic to the stance that we can merge your changes now and table the architectural decisions for now. What do you think?

Reply:
Understood! Makes sense! I'll make the switch!